csds-client/client/test_util.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+)
+
+// CaptureOutput captures the stdout for testing.
+func CaptureOutput(f func()) string {
+	reader, writer, err := os.Pipe()
+	if err != nil {
+		panic(err)
+	}
+	stdout := os.Stdout
+	stderr := os.Stderr
+	defer func() {
+		os.Stdout = stdout
+		os.Stderr = stderr
+	}()
+	os.Stdout = writer
+	os.Stderr = writer
+	out := make(chan string)
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		var buf bytes.Buffer
+		wg.Done()
+		io.Copy(&buf, reader)
+		out <- buf.String()
+	}()
+	wg.Wait()
+	f()
+	writer.Close()
+	return <-out
+}
+
+func ShouldEqualJSON(t *testing.T, s1, s2 string) bool {
```

Comment:
add some comment for this method?

Reply:
EqualJSONByptes as well.

Reply:
Actually these functions were written by @dio in the PR #16 . I just moved them outside to `test_util.go` to make them general to tests in different versions.

Reply:
I see. Still good to add some comments, I think.

Reply:
Added